parquet-column/src/main/java/org/apache/parquet/schema/Types.java

```diff
@@ -252,7 +252,12 @@ protected final THIS repetition(Type.Repetition repetition) {
      * @return this builder for method chaining
      */
     public THIS as(OriginalType type) {
-      this.originalType = type;
+      this.logicalTypeAnnotation = LogicalTypeAnnotation.fromOriginalType(type);
+      return self();
+    }
+
+    public THIS as(LogicalTypeAnnotation type) {
```

Comment:
This method breaks the fluent API of the builder as you need to use an outside factory method of the final type to create a LogicalTypeAnnotation. If it is practically feasible, I would suggest to refactor the LogicalTypeAnnotation API to fit more in the fluent API of Types.